### ISSUE TITLE
custom elements as pages (aka one off page templates)

### ISSUE THREAD

**Issue by thescientist13** [ProjectEvergreen/greenwood#170]
## Type of Change
- [ ] New Feature Request
- [ ] Documentation / Website
- [ ] Improvement / Suggestion
- [ ] Bug
- [x] Other (please clarify below)

## Summary
Currently to generate a page, Greenwood goes through the pages directory and builds out a route for every _.md_ file.
```
src/
  pages/
    about.md
    index.md
```

However, if you predominantly want to write a page in just JavaScript to generate HTML content dynamically you would have to do something like this in a page template instead:
```javascript
const postsData = [/* some array of  "post" objects */];
const postsList = postsData.map(item => {
  return html`<li>${item.title}</li>`;
})

...

render() {
   return html`
     <ul>
        ${ postsList }
     </ul>
   `
}
```

Basically _every_ page  needs a markdown file _and_ a template.  This is great for many to one (many pages, one template) but if your page is better authored as JavaScript, you still need to create two files.

## Details
So it would be great to see if maybe doing something like this would essentially treat _about.js_ as a Template by default?
```
src/
  pages/
    about.js
    index.md
```

This way, if you have one off or JavaScript heavy pages, you only have to make one file for it.  

Additionally, it would make sense that the templates folder would essentially just be a collection of layouts, which is most useful when stamping out many pages of the same structure (like a blog post)

**Comment by thescientist13** [comment]
This may end up falling under / becoming its own RFC.

**Comment by thescientist13** [comment]
Another great use case would be for projectevergreen.github.io.  For more "webapp" like projects, being able to use JS to author a page would be really nice.  This would allow the project to extract out [all the home page content](https://github.com/ProjectEvergreen/projectevergreen.github.io/blob/develop/src/templates/page-template.js#L44) into a custom element instead of stuffing the page-template full of code just to serialize some JS.

**Comment by thescientist13** [comment]
I think this might be invalid now that all users can build up their pages anyway they want now?

**Comment by thescientist13** [comment]
not needed, templates / pages are HTML first now in v0.10.0
